service/src/main/java/uk/gov/hmcts/reform/fpl/service/OthersService.java

```diff
@@ -31,6 +40,61 @@ public String buildOthersLabel(Others others) {
         return sb.toString();
     }
 
+    public Others prepareOthers(CaseData caseData) {
+        List<Element<Other>> others = new ArrayList<>();
+
+        caseData.getAllOthers().forEach(element -> {
+            if (element.getValue().containsConfidentialDetails()) {
+                Element<Other> confidentialOther = findConfidentialOther(caseData.getConfidentialOthers(), element);
+
+                others.add(element(element.getId(), addConfidentialDetails(confidentialOther, element)));
+            } else {
+                others.add(element);
+            }
+        });
+
+        return Others.builder()
+            .firstOther(getFirstOther(caseData.getConfidentialOthers(), others))
+            .additionalOthers(getAdditionalOthers(others))
+            .build();
+    }
+
+    private Element<Other> findConfidentialOther(List<Element<Other>> confidentialOthers, Element<Other> element) {
+        return confidentialOthers.stream()
+            .filter(confidentialOther -> confidentialOther.getId().equals(element.getId()))
+            .findFirst()
+            .orElse(element);
+    }
+
+    private Other addConfidentialDetails(Element<Other> confidentialOther, Element<Other> other) {
```

Comment:
We can remove this method and and use addConfidentialDetails method from model class

Reply:
this method in model class does not get correct telephone and address like here...